recipes/transformer-engine/001-cmake.patch

```diff
@@ -0,0 +1,11 @@
+--- a/build_tools/build_ext.py
++++ b/build_tools/build_ext.py
+@@ -56,7 +56,7 @@ class CMakeExtension(setuptools.Extension):
+             cmake_path,
+             "-B",
+             build_dir,
+-            f"-DPython_EXECUTABLE={sys.executable}",
++            "-DPython_EXECUTABLE=$PYTHON",
```

Comment:
Think we can do this in a more Pythonic way

```suggestion
+            "-DPython_EXECUTABLE={os.environ['PYTHON']}",
```

Also what happens when `sys.executable` is here?

Reply:
`sys.executable` led to this failure during build:

```
      CMake Error at /home/conda/staged-recipes/build_artifacts/transformer-engine_1745610886178/_build_env/share/cmake-4.0/Modules/FindPackageHandleStandardArgs.cmake:227 (message):
        Could NOT find Python (missing: Interpreter Development.Module) (found
        version "3.11.12")
```

Reply:
> Think we can do this in a more Pythonic way
No oddly, I get the same `missing: Interpreter Development.Module` when I do `os.environ['PYTHON']`; at least locally. Not sure why these would be different?